doc/samples/SqlConfigurableRetryLogic_SqlCommand.cs

```diff
@@ -0,0 +1,243 @@
+
+        // 2. Create a retry provider
+        var provider = SqlConfigurableRetryFactory.CreateExponentialRetryProvider(options);
+
+        // define the retrying event to report execution attempts
+        provider.Retrying += (object s, SqlRetryingEventArgs e) =>
+            {
+                int attempts = e.RetryCount + 1;
+                Console.ForegroundColor = ConsoleColor.Yellow;
+                Console.WriteLine($"attempt {attempts} - current delay time:{e.Delay} \n");
+                Console.ForegroundColor = ConsoleColor.DarkGray;
+                if (e.Exceptions[e.Exceptions.Count - 1] is SqlException ex)
+                {
+                    Console.WriteLine($"{ex.Number}-{ex.Message}\n");
+                }
+                else
+                {
+                    Console.WriteLine($"{e.Exceptions[e.Exceptions.Count - 1].Message}\n");
+                }
+
+                // It is not good practice to do time-consuming tasks inside the retrying event which blocks the running task.
+                // Use parallel programming patterns to mitigate it.
+                if (e.RetryCount == provider.RetryLogic.NumberOfTries - 1)
+                {
+                    Console.WriteLine("This is the last chance to execute the command before throwing the exception.");
+                    Console.WriteLine("Press Enter when you're ready:");
+                    Console.ReadLine();
+                    Console.WriteLine("continue ...");
+                }
+            };
+
+        // Open a general connection.
+        s_generalConnection.Open();
+
+        try
+        {
+            // Assume the database is creating and other services are going to connect to it.
+            RetryCommand(provider);
+        }
+        catch
+        {
+            // exception is thrown if connecting to the database isn't successful.
+            throw;
+        }
+    }
+
+    private static void ExecuteCommand(SqlConnection cn, string command)
+    {
+        using var cmd = cn.CreateCommand();
+        cmd.CommandText = command;
+        cmd.ExecuteNonQuery();
+    }
+
+    private static void FindActiveSessions(SqlConnection cnn, string dbName)
+    {
+        using var cmd = cnn.CreateCommand();
+        cmd.CommandText = "DECLARE @query NVARCHAR(max) = '';" + Environment.NewLine +
+            $"SELECT @query = @query + 'KILL ' + CAST(spid as varchar(50)) + ';' FROM sys.sysprocesses WHERE dbid = DB_ID('{dbName}')" + Environment.NewLine +
+            "SELECT @query AS Active_sessions;";
+        var reader = cmd.ExecuteReader();
+        if (reader.Read())
+        {
+            Console.ForegroundColor = ConsoleColor.Green;
+            Console.Write($">> Execute the '{reader.GetString(0)}' command in SQL Server to unblock the running task.");
+            Console.ResetColor();
+        }
+        reader.Close();
+    }
+// </Snippet1>
+// <Snippet2>
+    private static void RetryCommand(SqlRetryLogicBaseProvider provider)
```

Comment:
Can you add some comments to distinguish these `private static void RetryCommand(SqlRetryLogicBaseProvider provider)` ?

Reply:
You can find the description in the related [SqlCommand.xml](https://github.com/dotnet/SqlClient/blob/087fa76b2e4a5079012a6fe17a826c1d14e01af7/doc/snippets/Microsoft.Data.SqlClient/SqlCommand.xml#L2805) file. There is a common explanation for this series of samples. Feel free to suggest here If you need something more.